File: samples/ios-panthereyes-demo/ios/Sources/PantherEyesLabModels.swift
import Foundation

enum LabTextCategory: String {
    case cliCommand
    case agentPrompt
}

struct LabTextItem: Identifiable, Equatable {
    let id: UUID
    let title: String
    var value: String
    let originalValue: String
    let category: LabTextCategory
    let helperText: String?
    let isMonospaced: Bool

    init(
        id: UUID = UUID(),
        title: String,
        value: String,
        category: LabTextCategory,
        helperText: String? = nil,
        isMonospaced: Bool = true
    ) {
        self.id = id
        self.title = title
        self.value = value
        self.originalValue = value
        self.category = category
        self.helperText = helperText
        self.isMonospaced = isMonospaced
    }

    var isModified: Bool {
        value != originalValue
    }

    mutating func reset() {
        value = originalValue
    }
}

struct QuickLabStep: Identifiable, Equatable {
    let id: UUID
    let title: String
    let detail: String
    var isDone: Bool

    init(id: UUID = UUID(), title: String, detail: String, isDone: Bool = false) {
        self.id = id
        self.title = title
        self.detail = detail
        self.isDone = isDone
    }
}

struct CopyToastState: Equatable {
    let message: String
}

enum QuickDemoAction: String, CaseIterable, Identifiable {
    case validateConfig
    case runStaticScan
    case copyAgentPrompt

    var id: String { rawValue }

    var title: String {
        switch self {
        case .validateConfig:
            return "Validar Config"
        case .runStaticScan:
            return "Rodar Scan"
        case .copyAgentPrompt:
            return "Copiar Prompt"
        }
    }

    var subtitle: String {
        switch self {
        case .validateConfig:
            return "Copia comando de config validate"
        case .runStaticScan:
            return "Copia comando de scan estatico"
        case .copyAgentPrompt:
            return "Copia prompt do agente"
        }
    }

    var icon: String {
        switch self {
        case .validateConfig:
            return "checkmark.shield"
        case .runStaticScan:
            return "play.circle"
        case .copyAgentPrompt:
            return "bubble.left.and.text.bubble.right"
        }
    }
}

File: samples/android-panthereyes-demo/android/app/src/main/java/com/panthereyes/samples/androiddemo/lab/PantherEyesLabModels.kt
package com.panthereyes.samples.androiddemo.lab

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Build
import androidx.compose.material.icons.filled.Chat
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.ContentCopy
import androidx.compose.material.icons.filled.FlashOn
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.Star
import androidx.compose.ui.graphics.vector.ImageVector

enum class LabTextCategory {
    CLI_COMMAND,
    AGENT_PROMPT,
}

data class LabTextItem(
    val id: String,
    val title: String,
    val value: String,
    val originalValue: String,
    val category: LabTextCategory,
    val helperText: String? = null,
    val monospaced: Boolean = true,
) {
    val isModified: Boolean
        get() = value != originalValue
}

data class QuickLabStep(
    val id: String,
    val title: String,
    val detail: String,
    val isDone: Boolean = false,
)

enum class LabEnvironment {
    DEV,
    STAGING,
    PROD,
    ;

    val rawValue: String
        get() = name.lowercase()

    val displayName: String
        get() = name
}

enum class QuickDemoAction(
    val title: String,
    val subtitle: String,
    val icon: ImageVector,
) {
    VALIDATE_CONFIG(
        title = "Validar Config",
        subtitle = "Copia comando de config validate",
        icon = Icons.Filled.Settings,
    ),
    RUN_STATIC_SCAN(
        title = "Rodar Scan",
        subtitle = "Copia comando de scan estatico",
        icon = Icons.Filled.FlashOn,
    ),
    COPY_AGENT_PROMPT(
        title = "Copiar Prompt",
        subtitle = "Copia prompt do agente",
        icon = Icons.Filled.Chat,
    ),
}

data class LabCopyPayload(
    val text: String,
    val toastMessage: String,
)

enum class LabSectionMeta(
    val title: String,
    val icon: ImageVector,
) {
    HEADER("PantherEyes Android Demo", Icons.Filled.Star),
    DEMO("Demo Rapida", Icons.Filled.FlashOn),
    POLICY("Ambiente e Policy", Icons.Filled.Settings),
    CLI("Comandos CLI", Icons.Filled.Build),
    AGENT("Prompts do Agente", Icons.Filled.Chat),
    QUICK_LAB("Quick Lab", Icons.Filled.CheckCircle),
    NOTES("Notas do Sample", Icons.Filled.Info),
    COPY("Copiar", Icons.Filled.ContentCopy),
}
